src/tlv_utils.c

```diff
@@ -0,0 +1,252 @@
+#include "tlv_utils.h"
+
+#define DER_LONG_FORM_FLAG        0x80  // 8th bit set
+#define DER_FIRST_BYTE_VALUE_MASK 0x7f
+
+bool get_uint_from_tlv_data(const tlv_data_t *data, uint32_t *value) {
+    uint8_t size_diff;
+    uint8_t buffer[sizeof(uint32_t)];
+
+    if (data->length > sizeof(buffer)) {
+        PRINTF("Unexpectedly long value (%u bytes) for tag 0x%x\n", data->length, data->tag);
+        return false;
+    }
+    size_diff = sizeof(buffer) - data->length;
+    memset(buffer, 0, size_diff);
+    memcpy(buffer + size_diff, data->value, data->length);
+    *value = U4BE(buffer, 0);
+    return true;
+}
+
+bool get_uint8_t_from_tlv_data(const tlv_data_t *data, uint8_t *value) {
+    uint32_t tmp_value;
+    if (!get_uint_from_tlv_data(data, &tmp_value) || (tmp_value > UINT8_MAX)) {
+        return false;
+    }
+    *value = tmp_value;
+    return true;
+}
+
+bool get_cbuf_from_tlv_data(const tlv_data_t *data, cbuf_t *out, uint16_t min_size, uint16_t max_size) {
+    if (data->length < min_size) {
+        PRINTF("Expected at least %d bytes, found %D\n", min_size, data->length);
+        return false;
+    }
+    if (max_size != 0 && data->length > max_size) {
+        PRINTF("Expected at most %d bytes, found %D\n", max_size, data->length);
+        return false;
+    }
+    out->size = data->length;
+    out->bytes = data->value;
```

Comment:
The current helpers are 0 copy API
If the user wants to copy it can copy the result of the `get_cbuf_from_tlv_data` call or not use `get_cbuf_from_tlv_data`